app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,27 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    @user = User.find_by_id(params[:user_id]) # params[:user_id] is the id of the user
+    @posts = Post.where(author_id: params[:user_id])
+    @comments = []
+    @posts.map do |post|
+      @comments = get_comments(post, 'index')
+    end
```

Comment:
- [ ] In the user post page, you should use post.recent_comments method to render only 5 of the comment for each post as per the requirement